### PR TITLE
Give exoplanets a color based on their size

### PR DESCRIPTION
**Pull request by WeirdRubberDuck**
Adds a color to the exoplanets based on their size (Terrestrial, Super-Earth, Neptune-like or Gas giant and larger). 

Other additions include:
- adding a tag `exoplanet_planet` for the exoplanet
- adding descriptions to some of the scene graph nodes that ar eadded as part of the system (partly addresses issue #2864, but not fully)

The colors are inspired by the following graphic from NASA, with constants taken from this source: 
https://www.nasa.gov/image-article/sizes-of-known-exoplanets/
image

Here's an example for the KOI-351 system, which has planets of all 4 types
image

### CODE REVIEW

File: modules/exoplanets/exoplanetsmodule_lua.inl [hunk 1]
```diff
@@ -200,6 +200,25 @@ void createExoplanetSystem(const std::string& starName,
...
+    // @TODO (2024-09-16, emmbr) Compose a more extensive summary of the system,
+    // based on more data in the archive (like number of stars) and the plantes.
+    // Also include more data on the star itself (like spectral type)
+    float distanceToOurSystem = glm::length(starPosInParsec) *
+        distanceconstants::Parsec / distanceconstants::LightYear;
+
+    size_t nPlanets = system.planetNames.size();
+    const std::string starDescription = std::format(
+        "The star {} is the host star of an exoplanet system with {} {} that {}  "
+        "enough data to be visualized. It has a size of {:.2f} Solar radii and an "
```
Comment (alexanderbock) [inline modules/exoplanets/exoplanetsmodule_lua.inl:212]:
```suggestion
        "enough data to be visualized. It has a size of {:.2f} solar radii and an "
```

File: modules/exoplanets/exoplanetsmodule_lua.inl [hunk 2]
```diff
@@ -200,6 +200,25 @@ void createExoplanetSystem(const std::string& starName,
...
+        "effective temperature of {:.0f} Kelvin. The system is located at a "
+        "distance of {:.0f} Light years from Earth.",
```
Comment (alexanderbock) [inline modules/exoplanets/exoplanetsmodule_lua.inl:214]:
```suggestion
        "distance of {:.0f} light-years from Earth.",
```

File: modules/exoplanets/exoplanetsmodule_lua.inl [hunk 3]
```diff
@@ -335,15 +355,105 @@ void createExoplanetSystem(const std::string& starName,
         "}";
 
         std::string planetLayers = "";
+        std::string planetTypeDesc = "";
```
Comment (alexanderbock) [inline modules/exoplanets/exoplanetsmodule_lua.inl:358]:
```suggestion
        std::string planetLayers;
        std::string planetTypeDesc;
```

File: modules/exoplanets/exoplanetsmodule_lua.inl [hunk 4]
```diff
@@ -335,15 +355,105 @@ void createExoplanetSystem(const std::string& starName,
...
+
+        // Constant for different categories of sizes of planets (in Earth radii)
+        // Source: https://www.nasa.gov/image-article/sizes-of-known-exoplanets/
+        constexpr float MaxR_Terrestrial = 1.25f;
+        constexpr float MaxR_SuperEarth = 2.f;
+        constexpr float MaxR_NeptuneLike = 6.f;
```
Comment (alexanderbock) [inline modules/exoplanets/exoplanetsmodule_lua.inl:364]:
```suggestion
        constexpr float TerrestrialMaxR = 1.25f;
        constexpr float SuperEarthMaxR = 2.f;
        constexpr float NeptuneLikeMaxR = 6.f;
```
?

To make it fit our naming scheme

File: modules/exoplanets/exoplanetsmodule_lua.inl [hunk 5]
```diff
@@ -335,15 +355,105 @@ void createExoplanetSystem(const std::string& starName,
...
+            // Neptune-like
+            else if (rInMeter < MaxR_NeptuneLike * distanceconstants::EarthRadius) {
+                colorFromSize = glm::vec3(0.22f, 0.49f, 0.50f); // Blue
+                planetTypeDesc = NeptuneLikeDesc;
+            }
+            // Gas Giants (Saturn and Jupiter size, and much larger!)
+            else {
+                colorFromSize = glm::vec3(0.55f, 0.34f, 0.39f); // Wine red
+                planetTypeDesc = GasGiantDesc;
+            }
```
Comment (alexanderbock) [inline modules/exoplanets/exoplanetsmodule_lua.inl:412]:
```suggestion
            if (rInMeter < MaxR_Terrestrial * distanceconstants::EarthRadius) {
                // Terrestrial
                colorFromSize = glm::vec3(0.32f, 0.2f, 0.1f); // Brown
                planetTypeDesc = TerrestrialDesc;
            }
            else if (rInMeter < MaxR_SuperEarth * distanceconstants::EarthRadius) {
                // Super-Earths
                colorFromSize = glm::vec3(1.f, 0.76f, 0.65f); // Beige
                planetTypeDesc = SuperEarthDesc;
            }
            else if (rInMeter < MaxR_NeptuneLike * distanceconstants::EarthRadius) {
                // Neptune-like
                colorFromSize = glm::vec3(0.22f, 0.49f, 0.50f); // Blue
                planetTypeDesc = NeptuneLikeDesc;
            }
            else {
                // Gas Giants (Saturn and Jupiter size, and much larger!)
                colorFromSize = glm::vec3(0.55f, 0.34f, 0.39f); // Wine red
                planetTypeDesc = GasGiantDesc;
            }
```

File: modules/exoplanets/exoplanetsmodule_lua.inl [hunk 6]
```diff
@@ -335,15 +355,105 @@ void createExoplanetSystem(const std::string& starName,
...
+
+            std::string description = std::format(
```
Comment (alexanderbock) [inline modules/exoplanets/exoplanetsmodule_lua.inl:414]:
```suggestion
            const std::string description = std::format(
```

File: modules/exoplanets/exoplanetsmodule_lua.inl [hunk 7]
```diff
@@ -481,7 +598,9 @@ void createExoplanetSystem(const std::string& starName,
         "Tag = {'exoplanet_1au_ring'},"
         "GUI = {"
             "Name = '1 AU Size Comparison Circle',"
-            "Path = '" + guiPath + "'"
+            "Path = '" + guiPath + "',"
+            "Description = \"A circe with a radius of 1 Astronomical Unit. That is, its "
```
Comment (alexanderbock) [inline modules/exoplanets/exoplanetsmodule_lua.inl:602]:
```suggestion
            "Description = \"A circle with a radius of 1 Astronomical Unit. That is, its "
```

**Review by alexanderbock — APPROVED** [review]

File: modules/exoplanets/exoplanetsmodule_lua.inl [hunk 8]
```diff
@@ -200,6 +200,25 @@ void createExoplanetSystem(const std::string& starName,
             "}";
     }
 
+    // @TODO (2024-09-16, emmbr) Compose a more extensive summary of the system,
+    // based on more data in the archive (like number of stars) and the plantes.
```
Comment (WeirdRubberDuck) [inline modules/exoplanets/exoplanetsmodule_lua.inl:204]:
```suggestion
    // based on more data in the archive (like number of stars) and the planets.
```

File: modules/exoplanets/exoplanetsmodule_lua.inl [hunk 9]
```diff
@@ -226,7 +245,8 @@ void createExoplanetSystem(const std::string& starName,
         "Tag = {'exoplanet_system'},"
         "GUI = {"
             "Name = '" + sanitizedStarName + " (Star)',"
-            "Path = '" + guiPath + "'"
+            "Path = '" + guiPath + "',"
+            "Description = \"" +starDescription + "\""
```
Comment (WeirdRubberDuck) [inline modules/exoplanets/exoplanetsmodule_lua.inl:249]:
```suggestion
            "Description = \"" + starDescription + "\""
```

**Pull request merged**